File: src/main/java/no/novari/mapping/InstanceObjectCollectionMetadataMappingService.java
package no.novari.mapping;

import no.novari.model.dtos.InstanceObjectCollectionMetadataDto;
import no.novari.model.entities.InstanceObjectCollectionMetadata;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Service;

@Service
public class InstanceObjectCollectionMetadataMappingService {

    private final InstanceMetadataContentMappingService instanceMetadataContentMappingService;

    public InstanceObjectCollectionMetadataMappingService(
            @Lazy InstanceMetadataContentMappingService instanceMetadataContentMappingService
    ) {
        this.instanceMetadataContentMappingService = instanceMetadataContentMappingService;
    }

    public InstanceObjectCollectionMetadata toEntity(InstanceObjectCollectionMetadataDto instanceObjectCollectionMetadataDto) {
        return InstanceObjectCollectionMetadata
                .builder()
                .displayName(instanceObjectCollectionMetadataDto.getDisplayName())
                .objectMetadata(instanceMetadataContentMappingService.toEntity(
                        instanceObjectCollectionMetadataDto.getObjectMetadata()
                ))
                .key(instanceObjectCollectionMetadataDto.getKey())
                .build();
    }

    public InstanceObjectCollectionMetadataDto toDto(InstanceObjectCollectionMetadata instanceObjectCollectionMetadata) {
        return InstanceObjectCollectionMetadataDto
                .builder()
                .displayName(instanceObjectCollectionMetadata.getDisplayName())
                .objectMetadata(instanceMetadataContentMappingService.toDto(
                        instanceObjectCollectionMetadata.getObjectMetadata()
                ))
                .key(instanceObjectCollectionMetadata.getKey())
                .build();
    }

}

File: src/main/kotlin/no/novari/flyt/discovery/service/mapping/InstanceObjectCollectionMetadataMappingService.kt
package no.novari.flyt.discovery.service.mapping

import no.novari.flyt.discovery.service.model.dtos.InstanceObjectCollectionMetadataDto
import no.novari.flyt.discovery.service.model.entities.InstanceObjectCollectionMetadata
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Service

@Service
class InstanceObjectCollectionMetadataMappingService(
    @param:Lazy private val instanceMetadataContentMappingService: InstanceMetadataContentMappingService,
) {
    fun toEntity(
        instanceObjectCollectionMetadataDto: InstanceObjectCollectionMetadataDto,
    ): InstanceObjectCollectionMetadata =
        InstanceObjectCollectionMetadata(
            displayName = requireNotNull(instanceObjectCollectionMetadataDto.displayName),
            objectMetadata =
                instanceMetadataContentMappingService.toEntity(
                    requireNotNull(instanceObjectCollectionMetadataDto.objectMetadata),
                ),
            key = requireNotNull(instanceObjectCollectionMetadataDto.key),
        )

    fun toDto(
        instanceObjectCollectionMetadata: InstanceObjectCollectionMetadata,
    ): InstanceObjectCollectionMetadataDto =
        InstanceObjectCollectionMetadataDto(
            displayName = requireNotNull(instanceObjectCollectionMetadata.displayName),
            objectMetadata =
                instanceMetadataContentMappingService.toDto(
                    requireNotNull(instanceObjectCollectionMetadata.objectMetadata),
                ),
            key = requireNotNull(instanceObjectCollectionMetadata.key),
        )
}
